diff --git a/stlport/STLport-4.5.patch b/stlport/STLport-4.5.patch
index 27ff2ef..8476053 100644
--- a/stlport/STLport-4.5.patch
+++ b/stlport/STLport-4.5.patch
@@ -137,7 +137,7 @@
 +#
 +#
 +LINK=ar cr
-+DYN_LINK=${CXX} -Wl,-rpath,'$$ORIGIN' ${PTHREAD_LIBS} --fexceptions -shared -o
++DYN_LINK=${CXX} -Wl,-rpath,'$$ORIGIN' ${PTHREAD_LIBS} -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=so
@@ -210,7 +210,7 @@
 +#
 +#
 +LINK=ar crv
-+DYN_LINK=$(CXX) --fexceptions -shared -o
++DYN_LINK=$(CXX) -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=dll
@@ -449,7 +449,7 @@
 +#
 +#
 +LINK=ar cr
-+DYN_LINK=$(CXX) -Wl,-rpath,'$$ORIGIN' --fexceptions -shared -o
++DYN_LINK=$(CXX) -Wl,-rpath,'$$ORIGIN' -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=so
diff --git a/stlport/systemstlguards/preextstl.h b/stlport/systemstlguards/preextstl.h
index d0ee030..5ff8668 100644
--- a/stlport/systemstlguards/preextstl.h
+++ b/stlport/systemstlguards/preextstl.h
@@ -48,7 +48,6 @@
 #       if defined(_GNUC__)
 #           pragma GCC visibility push(default)
 #       endif
-#       include _STLP_NATIVE_HEADER(exception_defines.h)
 #       include _STLP_NATIVE_HEADER(limits)
 #       include _STLP_NATIVE_HEADER(memory)
 #       include _STLP_NATIVE_HEADER(exception)
diff --git a/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx b/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
index 044dfc5..1d28d05 100644
--- a/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
+++ b/bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
@@ -32,9 +32,10 @@
 #include <stdio.h>
 #include <string.h>
 #include <dlfcn.h>
-#include <cxxabi.h>
 #include <hash_map>
 
+#include <cxxabi.h>
+
 #include <rtl/strbuf.hxx>
 #include <rtl/ustrbuf.hxx>
 #include <osl/diagnose.h>
@@ -47,7 +48,6 @@
 
 #include "share.hxx"
 
-
 using namespace ::std;
 using namespace ::osl;
 using namespace ::rtl;
