From 820efe20886e41fdea8353f54e8a54bba7fccd84 Mon Sep 17 00:00:00 2001
From: rpmbuild <rpmbuild@fedoraproject.org>
Date: Thu, 20 Feb 2014 19:44:49 +0100
Subject: [PATCH] never run autogen.sh

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 206f5ed..14620da 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -14,7 +14,7 @@ MAKECMDGOALS?=all
 SHELL := @SHELL_BASH@
 SRCDIR := @SRC_ROOT@
 BUILDDIR := @BUILDDIR@
-GIT_BUILD := $(if $(wildcard $(SRCDIR)/.git),T)
+GIT_BUILD :=
 
 # Run autogen.sh if needed and force make to restart itself.
 # ... but there are several cases where we do not want to run
-- 
1.8.5.3

