From eca998dc1dc874415794a27336d5db1679ef69b8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mat=C3=BA=C5=A1=20Kukan?= <matus.kukan@gmail.com>
Date: Sat, 17 Dec 2011 13:29:50 +0100
Subject: [PATCH] gcc-trunk: avoid confusion

Signed-off-by: David Tardon <dtardon@redhat.com>
---
 svtools/source/graphic/graphic.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/svtools/source/graphic/graphic.hxx b/svtools/source/graphic/graphic.hxx
index 7b4f5e0..a1efca0 100644
--- a/svtools/source/graphic/graphic.hxx
+++ b/svtools/source/graphic/graphic.hxx
@@ -58,7 +58,7 @@ public:
     Graphic();
     ~Graphic() throw();
 
-    using unographic::GraphicDescriptor::init;
+    using ::unographic::GraphicDescriptor::init;
     void init( const ::Graphic& rGraphic ) throw();
 
     static const ::Graphic* getImplementation( const ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface >& rxIFace ) throw();
-- 
1.7.7.5

