From 9745fccacaa88bded763cc31c9aa7c04a5186a48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 10 Feb 2011 09:51:46 +0000
Subject: [PATCH] Resolves: rhbz#676539 handle missing pWindows from xWindows

---
 framework/source/layoutmanager/layoutmanager.cxx |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/framework/source/layoutmanager/layoutmanager.cxx b/framework/source/layoutmanager/layoutmanager.cxx
index a67c71d..2905f03 100644
--- a/framework/source/layoutmanager/layoutmanager.cxx
+++ b/framework/source/layoutmanager/layoutmanager.cxx
@@ -130,13 +130,13 @@ bool lcl_checkUIElement(const Reference< XUIElement >& xUIElement,css::awt::Rect
         _rPosSize = _xWindow->getPosSize();
 
         Window* pWindow = VCLUnoHelper::GetWindow( _xWindow );
-        if ( pWindow->GetType() == WINDOW_TOOLBOX )
+        if ( pWindow && pWindow->GetType() == WINDOW_TOOLBOX )
         {
             ::Size aSize = ((ToolBox*)pWindow)->CalcWindowSizePixel( 1 );
             _rPosSize.Width = aSize.Width();
             _rPosSize.Height = aSize.Height();
         }
-    } // if ( xUIElement.is() )
+    }
     return bRet;
 }
 
@@ -1007,9 +1007,9 @@ void LayoutManager::implts_createAddonsToolBars()
                     // Set generic title for add-on toolbar
                     SolarMutexGuard aGuard;
                     Window* pWindow = VCLUnoHelper::GetWindow( xWindow );
-                    if ( pWindow->GetText().Len() == 0 )
+                    if ( pWindow && pWindow->GetText().Len() == 0 )
                         pWindow->SetText( aGenericAddonTitle );
-                    if ( pWindow->GetType() == WINDOW_TOOLBOX )
+                    if ( pWindow && pWindow->GetType() == WINDOW_TOOLBOX )
                     {
                         ToolBox* pToolbar = (ToolBox *)pWindow;
                         pToolbar->SetMenuType();
-- 
1.7.4

