From e65850b4411c59a5c664b002c6ab4ae385fd2261 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 13 Dec 2010 12:48:47 +0000
Subject: [PATCH] valgrind: don't leave an evil thread running after main is completed

---
 configmgr/source/components.cxx |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/configmgr/source/components.cxx b/configmgr/source/components.cxx
index e6cfb10..44c2d32 100644
--- a/configmgr/source/components.cxx
+++ b/configmgr/source/components.cxx
@@ -595,7 +595,10 @@ Components::Components(
     RTL_LOGFILE_TRACE_AUTHOR("configmgr", "sb", "end parsing");
 }
 
-Components::~Components() {}
+Components::~Components()
+{
+    flushModifications();
+}
 
 void Components::parseFileLeniently(
     FileParser * parseFile, rtl::OUString const & url, int layer, Data & data,
-- 
1.7.4.1

