From e5bbb64626106bdfceea47b7b7c580172d5ae420 Mon Sep 17 00:00:00 2001
From: Lionel Elie Mamane <lionel@mamane.lu>
Date: Mon, 13 Feb 2012 17:36:23 +0100
Subject: [PATCH 1/2] fdo#43399 hidden radio button should also gets unset

Signed-off-by: Michael Meeks <michael.meeks@suse.com>
---
 extensions/source/abpilot/typeselectionpage.cxx |   12 +++++-------
 1 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/extensions/source/abpilot/typeselectionpage.cxx b/extensions/source/abpilot/typeselectionpage.cxx
index f78878f..020e3a0 100644
--- a/extensions/source/abpilot/typeselectionpage.cxx
+++ b/extensions/source/abpilot/typeselectionpage.cxx
@@ -112,9 +112,13 @@ namespace abp
         m_aAllTypes.push_back( ButtonItem( &m_aOE, AST_OE, bWithMozilla && !bUnx ) );
         m_aAllTypes.push_back( ButtonItem( &m_aOther, AST_OTHER, true ) );
 
-        bool bFirstVisible = true;
         Link aTypeSelectionHandler = LINK(this, TypeSelectionPage, OnTypeSelected );
         const Size aSpacing( LogicToPixel( Size( 0, 3 ), MAP_APPFONT ) );
+        if ( ! m_aAllTypes.empty() )
+        {
+            ButtonItem aItem = m_aAllTypes[0];
+            aItem.m_pItem->SetStyle( aItem.m_pItem->GetStyle() | WB_GROUP );
+        }
         for ( ::std::vector< ButtonItem >::const_iterator loop = m_aAllTypes.begin();
               loop != m_aAllTypes.end(); ++loop )
         {
@@ -127,12 +131,6 @@ namespace abp
                 aTopLeft.Y() += aItemSize.Height() + aSpacing.Height();
                 aItem.m_pItem->SetClickHdl( aTypeSelectionHandler );
                 aItem.m_pItem->Show();
-
-                if ( bFirstVisible )
-                {
-                    aItem.m_pItem->SetStyle( aItem.m_pItem->GetStyle() | WB_GROUP );
-                    bFirstVisible = false;
-                }
             }
         }
     }
-- 
1.7.7.6

