diff --git a/stlport/STLport-4.5.patch b/stlport/STLport-4.5.patch
index 27ff2ef..8476053 100644
--- a/stlport/STLport-4.5.patch
+++ b/stlport/STLport-4.5.patch
@@ -137,7 +137,7 @@
 +#
 +#
 +LINK=ar cr
-+DYN_LINK=${CXX} -Wl,-rpath,'$$ORIGIN' ${PTHREAD_LIBS} --fexceptions -shared -o
++DYN_LINK=${CXX} -Wl,-rpath,'$$ORIGIN' ${PTHREAD_LIBS} -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=so
@@ -210,7 +210,7 @@
 +#
 +#
 +LINK=ar crv
-+DYN_LINK=$(CXX) --fexceptions -shared -o
++DYN_LINK=$(CXX) -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=dll
@@ -449,7 +449,7 @@
 +#
 +#
 +LINK=ar cr
-+DYN_LINK=$(CXX) -Wl,-rpath,'$$ORIGIN' --fexceptions -shared -o
++DYN_LINK=$(CXX) -Wl,-rpath,'$$ORIGIN' -fexceptions -shared -o
 +
 +OBJEXT=o
 +DYNEXT=so
diff --git a/stlport/systemstlguards/preextstl.h b/stlport/systemstlguards/preextstl.h
index d0ee030..5ff8668 100644
--- a/stlport/systemstlguards/preextstl.h
+++ b/stlport/systemstlguards/preextstl.h
@@ -48,7 +48,11 @@
 #       if defined(_GNUC__)
 #           pragma GCC visibility push(default)
 #       endif
-#       include _STLP_NATIVE_HEADER(exception_defines.h)
+#       if (__GNUC__ >= 5 || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 6)))
+#           include _STLP_NATIVE_HEADER(bits/exception_defines.h)
+#       else
+#           include _STLP_NATIVE_HEADER(exception_defines.h)
+#       endif
 #       include _STLP_NATIVE_HEADER(limits)
 #       include _STLP_NATIVE_HEADER(memory)
 #       include _STLP_NATIVE_HEADER(exception)
