From 1a3533da32df2217e389c3f7c97cb642427d864b Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Fri, 13 Jan 2012 11:25:49 +0100
Subject: [PATCH 3/8] Fixed apparent typos.

---
 basic/source/sbx/sbxvalue.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/basic/source/sbx/sbxvalue.cxx b/basic/source/sbx/sbxvalue.cxx
index 6d9d718..6448158 100644
--- a/basic/source/sbx/sbxvalue.cxx
+++ b/basic/source/sbx/sbxvalue.cxx
@@ -1319,7 +1319,7 @@ Lbl_OpIsDouble:
                             SetError( SbxERR_NOTIMP );
                     }
                     // #45465 Date needs with "+" a special handling: forces date type
-                    if( Op == SbxPLUS && (GetType() == SbxDATE || rOp.GetType() == SbxDATE )
+                    if( eOp == SbxPLUS && (GetType() == SbxDATE || rOp.GetType() == SbxDATE ) )
                         aL.eType = SbxDATE;
 
                 }
-- 
1.7.7.6

