diff -up ./configure.ac.tv ./configure.ac
--- ./configure.ac.tv	2020-01-30 11:51:51.332400167 +0100
+++ ./configure.ac	2020-01-30 11:51:56.775508000 +0100
@@ -10034,7 +10034,7 @@ if test "$with_system_qrcodegen" = "yes"
     AC_MSG_RESULT([external])
     SYSTEM_QRCODEGEN=TRUE
     AC_LANG_PUSH([C++])
-    AC_CHECK_HEADER(qrcodegen/QrCode.hpp, [],
+    AC_CHECK_HEADER(qrcodegencpp/QrCode.hpp, [],
         [AC_MSG_ERROR(qrcodegen headers not found.)], [#include <stdexcept>])
     AC_CHECK_LIB([qrcodegencpp], [main], [:],
         [ AC_MSG_ERROR(qrcodegen C++ library not found.) ], [])
diff -up ./configure.tv ./configure
--- ./configure.tv	2020-01-30 11:51:53.173436638 +0100
+++ ./configure	2020-01-30 11:52:06.101692758 +0100
@@ -33732,7 +33732,7 @@ ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS
 ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
-    ac_fn_cxx_check_header_compile "$LINENO" "qrcodegen/QrCode.hpp" "ac_cv_header_qrcodegen_QrCode_hpp" "#include <stdexcept>
+    ac_fn_cxx_check_header_compile "$LINENO" "qrcodegencpp/QrCode.hpp" "ac_cv_header_qrcodegen_QrCode_hpp" "#include <stdexcept>
 "
 if test "x$ac_cv_header_qrcodegen_QrCode_hpp" = xyes; then :
 
