--- a/solenv/inc/postset.mk	2010-09-29 12:55:57.000000000 +0100
+++ b/solenv/inc/postset.mk	2010-09-29 12:58:29.000000000 +0100
@@ -33,15 +33,13 @@
 # Complete list of all supported ISO codes
 completelangiso=af \
 ar \
-as-IN \
+as \
 be-BY \
 bo \
 bg \
 br \
 brx \
 bn \
-bn-BD \
-bn-IN \
 bs \
 ca \
 cs \
@@ -65,7 +63,6 @@
 gd \
 gl \
 gu \
-gu-IN \
 he \
 hi \
 hr \
@@ -91,8 +88,8 @@
 mn \
 mni \
 ms \
-ml-IN \
-mr-IN \
+ml \
+mr \
 my \
 ne \
 nb \
@@ -102,7 +99,7 @@
 ns \
 om \
 oc \
-or-IN \
+or \
 pap \
 pa-IN \
 pl \
@@ -126,16 +123,16 @@
 sv \
 sw \
 sw-TZ \
-te-IN \
-ti-ER \
-ta-IN \
+te \
+ti \
+ta \
 th \
 tn \
 tr \
 ts \
 tg \
 ug \
-ur-IN \
+ur \
 uk \
 uz \
 ve \
--- a/configure.in
+++ b/configure.in
@@ -3815,12 +3815,6 @@ if test -n "$with_system_cppunit" -o -n "$with_system_libs" && \
     SYSTEM_CPPUNIT=YES
     # might work for earlier, too but go sure
     PKG_CHECK_MODULES( CPPUNIT, cppunit >= 1.12.0 )
-    AC_MSG_CHECKING([STL compatibility])
-    if test "$WITH_STLPORT" != "no"; then
-	AC_MSG_ERROR([to use system cppunit you need to use --without-stlport])
-    else
-	AC_MSG_RESULT([OK])	
-    fi
 else
     AC_MSG_RESULT([internal])
     SYSTEM_CPPUNIT=NO
