--- a/shell/Module_shell.mk	2013-02-13 15:48:02.880128705 +0000
+++ b/shell/Module_shell.mk	2013-02-13 16:37:33.766353120 +0000
@@ -32,14 +32,6 @@
 	Library_localebe \
 ))
 
-ifeq ($(OS),LINUX)
-ifeq ($(ENABLE_GIO),TRUE)
-$(eval $(call gb_Module_add_targets,shell,\
-	Library_losessioninstall \
-))
-endif
-endif
-
 ifeq ($(ENABLE_GCONF),TRUE)
 $(eval $(call gb_Module_add_targets,shell,\
 	Library_gconfbe \
diff --git a/Repository.mk b/Repository.mk
index d0de1d8..526fea3 100644
--- a/Repository.mk
+++ b/Repository.mk
@@ -229,7 +229,6 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,GNOME, \
     $(if $(ENABLE_GTK3),vclplug_gtk3) \
     $(if $(ENABLE_GCONF),gconfbe1) \
     $(if $(ENABLE_SYSTRAY_GTK),qstart_gtk) \
-    $(if $(ENABLE_GIO),losessioninstall) \
     $(if $(ENABLE_GIO),ucpgio1) \
     $(if $(ENABLE_GNOMEVFS),ucpgvfs1) \
 ))
diff --git a/postprocess/Rdb_services.mk b/postprocess/Rdb_services.mk
index 936aac2..7d26eb4 100644
--- a/postprocess/Rdb_services.mk
+++ b/postprocess/Rdb_services.mk
@@ -197,7 +197,6 @@ $(eval $(call gb_Rdb_add_components,services,\
 		shell/source/backends/gconfbe/gconfbe1 \
 	) \
 	$(if $(ENABLE_GIO), \
-		shell/source/sessioninstall/losessioninstall \
 		ucb/source/ucp/gio/ucpgio \
 	) \
 	$(if $(ENABLE_GNOMEVFS), \
