From 2ed57bc3ba020ebc6aec933445ae0b274685499e Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 3 Feb 2014 21:41:00 +0100
Subject: [PATCH] installation fix

---
 solenv/bin/modules/installer/worker.pm | 1 -
 1 file changed, 1 deletion(-)

diff --git a/solenv/bin/modules/installer/worker.pm b/solenv/bin/modules/installer/worker.pm
index aab676f..0e6f67d 100644
--- a/solenv/bin/modules/installer/worker.pm
+++ b/solenv/bin/modules/installer/worker.pm
@@ -96,7 +96,6 @@ sub create_installation_directory
 
     $installdir = installer::systemactions::create_directories("install", $languageref);
     installer::logger::print_message( "... creating installation set in $installdir ...\n" );
-    remove_old_installation_sets($installdir);
     my $inprogressinstalldir = $installdir . "_inprogress";
     installer::systemactions::rename_directory($installdir, $inprogressinstalldir);
     $installdir = $inprogressinstalldir;
-- 
1.8.4.2

