--- a/scp2/source/ooo/common_brand.scp
+++ b/scp2/source/ooo/common_brand.scp
@@ -1252,6 +1252,7 @@
     Section = "Bootstrap";
     Key = "URE_MORE_JAVA_CLASSPATH_URLS";
     Value = "";
+    ValueList1 = "$ORIGIN/../ure-link/lib";
 #ifdef SYSTEM_HSQLDB
     ValueList1 = HSQLDB_JAR;
 #endif
--- a/odk/CustomTarget_javadoc.mk
+++ b/odk/CustomTarget_javadoc.mk
@@ -40,7 +40,6 @@ $(call gb_CustomTarget_get_workdir,odk/docs/java/ref)/javadoc_log.txt : $(call g
 		-linkoffline ../../common/reg $(SRCDIR)/odk/pack/gendocu/uno \
 		-linkoffline http://java.sun.com/j2se/1.5/docs/api $(SRCDIR)/odk/pack/gendocu/java \
 		$(odk_JAVAPACKAGES) \
-		$(if $(JAVADOCISGJDOC),,-notimestamp) \
 		> $@
 
 # vim: set noet sw=4 ts=4:
