--- vcl/inc/vcl/settings.hxx	2012-03-29 08:39:08.838789585 +0100
+++ vcl/inc/vcl/settings.hxx	2012-03-29 08:41:26.375587207 +0100
@@ -1144,24 +1144,24 @@
 // - AllSettings -
 // ---------------
 
-#define SETTINGS_MACHINE            ((sal_uLong)0x00000001)
-#define SETTINGS_MOUSE              ((sal_uLong)0x00000002)
-#define SETTINGS_KEYBOARD           ((sal_uLong)0x00000004)
-#define SETTINGS_STYLE              ((sal_uLong)0x00000008)
-#define SETTINGS_MISC               ((sal_uLong)0x00000010)
-#define SETTINGS_SOUND              ((sal_uLong)0x00000020)
-#define SETTINGS_NOTIFICATION       ((sal_uLong)0x00000040)
-#define SETTINGS_HELP               ((sal_uLong)0x00000080)
-#define SETTINGS_INTERNATIONAL      ((sal_uLong)0x00000100) /* was for class International, has no effect anymore */
-#define SETTINGS_LOCALE             ((sal_uLong)0x00000200)
-#define SETTINGS_UILOCALE           ((sal_uLong)0x00000400)
-#define SETTINGS_ALLSETTINGS        (SETTINGS_MACHINE |\
-                                     SETTINGS_MOUSE | SETTINGS_KEYBOARD |\
-                                     SETTINGS_STYLE | SETTINGS_MISC |\
-                                     SETTINGS_SOUND | SETTINGS_NOTIFICATION |\
-                                     SETTINGS_HELP |\
-                                     SETTINGS_LOCALE | SETTINGS_UILOCALE )
-#define SETTINGS_IN_UPDATE_SETTINGS ((sal_uLong)0x00000800)	  // this flag indicates that the data changed event was created
+const sal_uLong SETTINGS_MACHINE =          ((sal_uLong)0x00000001);
+const sal_uLong SETTINGS_MOUSE   =          ((sal_uLong)0x00000002);
+const sal_uLong SETTINGS_KEYBOARD =         ((sal_uLong)0x00000004);
+const sal_uLong SETTINGS_STYLE   =          ((sal_uLong)0x00000008);
+const sal_uLong SETTINGS_MISC    =          ((sal_uLong)0x00000010);
+const sal_uLong SETTINGS_SOUND   =          ((sal_uLong)0x00000020);
+const sal_uLong SETTINGS_NOTIFICATION =     ((sal_uLong)0x00000040);
+const sal_uLong SETTINGS_HELP    =          ((sal_uLong)0x00000080);
+const sal_uLong SETTINGS_INTERNATIONAL =    ((sal_uLong)0x00000100); /* was for class International, has no effect anymore */
+const sal_uLong SETTINGS_LOCALE  =          ((sal_uLong)0x00000200);
+const sal_uLong SETTINGS_UILOCALE =         ((sal_uLong)0x00000400);
+const sal_uLong SETTINGS_ALLSETTINGS =      (SETTINGS_MACHINE |\
+                            SETTINGS_MOUSE | SETTINGS_KEYBOARD |\
+                            SETTINGS_STYLE | SETTINGS_MISC |\
+                            SETTINGS_SOUND | SETTINGS_NOTIFICATION |\
+                            SETTINGS_HELP |\
+                            SETTINGS_LOCALE | SETTINGS_UILOCALE );
+const sal_uLong SETTINGS_IN_UPDATE_SETTINGS = ((sal_uLong)0x00000800);	  // this flag indicates that the data changed event was created
                                                           // in Windows::UpdateSettings probably because of a global
                                                           // settings changed
 
--- fpicker/source/unx/kde4/KDE4FilePicker.cxx	2012-03-29 08:38:59.298664830 +0100
+++ fpicker/source/unx/kde4/KDE4FilePicker.cxx	2012-03-29 08:39:17.772906408 +0100
@@ -55,12 +55,6 @@
 /* ********* Hack, but needed because of conflicting types... */
 #define Region QtXRegion
 
-//kde has an enum that uses this...OO does too
-#define LO_SETTINGS_MOUSE SETTINGS_MOUSE
-#undef SETTINGS_MOUSE
-#define LO_SETTINGS_LOCALE SETTINGS_LOCALE
-#undef SETTINGS_LOCALE
-
 #include <kfiledialog.h>
 #include <kwindowsystem.h>
 #include <kapplication.h>
--- vcl/unx/kde4/KDEXLib.cxx	2012-04-18 15:51:31.996050751 +0100
+++ vcl/unx/kde4/KDEXLib.cxx	2012-04-18 15:52:23.119606979 +0100
@@ -57,7 +57,7 @@
 
 #include <stdio.h>
 
-#if QT_VERSION >= QT_VERSION_CHECK( 4, 8, 1 )
+#if QT_VERSION >= QT_VERSION_CHECK( 4, 9, 0 )
 #define QT_UNIX_EVENT_LOOP_SUPPORT
 #ifdef KDE_HAVE_GLIB
 #define GLIB_EVENT_LOOP_SUPPORT
