diff -r 98d427d68ce0 sw/source/core/txtnode/thints.cxx
--- a/sw/source/core/txtnode/thints.cxx	Thu Nov 04 14:44:14 2010 +0000
+++ b/sw/source/core/txtnode/thints.cxx	Sat Nov 27 16:03:40 2010 +0100
@@ -2102,7 +2102,7 @@
         const SwTxtAttr* const pAutoStyle(i_rAttrSpan.second);
         SfxItemIter aIter(m_rAttrSet);
         const SfxPoolItem* pItem(aIter.GetCurItem());
-        while (true)
+        while (pItem)
         {
             const USHORT nWhich(pItem->Which());
             if (CharFmt::IsItemIncluded(nWhich, pAutoStyle))
@@ -2170,7 +2170,7 @@
     o_rClearIds.reserve(i_rAttrSet.Count());
     SfxItemIter aIter(i_rAttrSet);
     const SfxPoolItem* pItem(aIter.GetCurItem());
-    while (true)
+    while (pItem)
     {
         o_rClearIds.push_back(pItem->Which());
 
