diff -r 2da10c20ba01 sw/source/core/layout/trvlfrm.cxx
--- a/sw/source/core/layout/trvlfrm.cxx	Mon Feb 08 10:49:42 2010 +0000
+++ b/sw/source/core/layout/trvlfrm.cxx	Tue Feb 16 10:35:18 2010 +0000
@@ -2064,9 +2064,11 @@
 	if ( pStartFrm->IsInFly() )
 	{
         const SwAnchoredObject* pObj = pStartFrm->FindFlyFrm();
-        aSortObjs.Insert( *(const_cast<SwAnchoredObject*>(pObj)) );
+        ASSERT( pObj, "No Start Object." );
+        if (pObj) aSortObjs.Insert( *(const_cast<SwAnchoredObject*>(pObj)) );
         const SwAnchoredObject* pObj2 = pEndFrm->FindFlyFrm();
-        aSortObjs.Insert( *(const_cast<SwAnchoredObject*>(pObj2)) );
+        ASSERT( pObj2, "No Start Object." );
+        if (pObj2) aSortObjs.Insert( *(const_cast<SwAnchoredObject*>(pObj2)) );
 	}
 
 	//Fall 4: Tabellenselection
