Index: source/filter/ww8/wrtw8sty.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/wrtw8sty.cxx,v
retrieving revision 1.49.44.2
diff -u -r1.49.44.2 wrtw8sty.cxx
--- openoffice.org.orig/sw/source/filter/ww8/wrtw8sty.cxx	12 Dec 2008 11:11:33 -0000	1.49.44.2
+++ openoffice.org/sw/source/filter/ww8/wrtw8sty.cxx	9 Apr 2009 12:02:50 -0000
@@ -621,7 +621,14 @@
 
     ShortToSVBT16( 400, &maWW8_FFN[2] );        // weiss ich nicht besser
                                                 // 400 == FW_NORMAL (windows.h)
-    maWW8_FFN[4] = sw::ms::rtl_TextEncodingToWinCharset(eChrSet);
+                                                //
+    //#i61927# For unicode fonts like Arial Unicode, Word 97+ sets the chs
+    //to SHIFTJIS presumably to capture that it's a multi-byte encoding font
+    //but Word95 doesn't do this, and sets it to 0 (ANSI), so we should do the
+    //same
+    maWW8_FFN[4] = bWrtWW8 ? 
+        sw::ms::rtl_TextEncodingToWinCharset(eChrSet) :
+        rtl_getBestWindowsCharsetFromTextEncoding(eChrSet);
 
     if (mbAlt)
         maWW8_FFN[5] = static_cast< BYTE >(msFamilyNm.Len() + 1);
