diff -up libreoffice-3.3.2.2/mdds/makefile.mk.dt libreoffice-3.3.2.2/mdds/makefile.mk
--- libreoffice-3.3.2.2/mdds/makefile.mk.dt	2011-03-24 16:59:53.237493218 +0100
+++ libreoffice-3.3.2.2/mdds/makefile.mk	2011-03-24 17:26:51.098221650 +0100
@@ -38,7 +38,8 @@ TARGET=mdds
 
 TARFILE_NAME=mdds_0.3.0
 TARFILE_MD5=cf8a6967f7de535ae257fa411c98eb88
-PATCH_FILES=
+PATCH_FILES= mdds.add-missing-link.patch \
+             mdds.do-not-insert-new-node.patch
 
 CONFIGURE_DIR=
 CONFIGURE_ACTION=
