--- vbahelper/inc/vbahelper/vbacollectionimpl.hxx	2011-01-28 20:27:51.507604173 +0000
+++ vbahelper/inc/vbahelper/vbacollectionimpl.hxx	2011-01-28 20:28:26.230045727 +0000
@@ -238,7 +238,7 @@
 
 // including a HelperInterface implementation
 template< typename Ifc1 >
-class ScVbaCollectionBase : public InheritedHelperInterfaceImpl< Ifc1 >
+class VBAHELPER_DLLPUBLIC ScVbaCollectionBase : public InheritedHelperInterfaceImpl< Ifc1 >
 {
 typedef InheritedHelperInterfaceImpl< Ifc1 > BaseColBase;
 protected:
--- solenv/inc/unxgcc.mk	2011-01-31 08:57:25.476403469 +0000
+++ solenv/inc/unxgcc.mk	2011-01-31 08:57:51.556952522 +0000
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # -fpermissive should be removed as soon as possible
-CFLAGSCXX= -pipe $(ARCH_FLAGS)
+CFLAGSCXX= -pipe $(ARCH_FLAGS) -fno-implicit-templates
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
 CFLAGSCXX += -fvisibility-inlines-hidden
 .ENDIF # "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
