diff -r bec9b2b2628f -r 47329487e9cf reportbuilder/java/com/sun/star/report/pentaho/loader/InputRepositoryLoader.java
--- a/reportbuilder/java/com/sun/star/report/pentaho/loader/InputRepositoryLoader.java	Mon Nov 09 14:52:58 2009 +0100
+++ b/reportbuilder/java/com/sun/star/report/pentaho/loader/InputRepositoryLoader.java	Wed Nov 11 10:36:44 2009 +0100
@@ -35,6 +35,7 @@
 
 import org.pentaho.reporting.libraries.resourceloader.ResourceData;
 import org.pentaho.reporting.libraries.resourceloader.ResourceKey;
+import org.pentaho.reporting.libraries.resourceloader.ResourceException;
 import org.pentaho.reporting.libraries.resourceloader.ResourceKeyCreationException;
 import org.pentaho.reporting.libraries.resourceloader.ResourceLoader;
 import org.pentaho.reporting.libraries.resourceloader.ResourceLoadingException;
@@ -169,4 +170,19 @@
     {
         return resourceManager;
     }
+
+    public boolean isSupportedDeserializer(String string)
+    {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    public String serialize(ResourceKey rk, ResourceKey rk1) throws ResourceException
+    {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    public ResourceKey deserialize(ResourceKey rk, String string) throws ResourceKeyCreationException
+    {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
 }
