diff -r e08d3a985a67 vcl/unx/gtk/window/gtkframe.cxx
--- a/vcl/unx/gtk/window/gtkframe.cxx	Fri Aug 06 13:25:47 2010 +0100
+++ b/vcl/unx/gtk/window/gtkframe.cxx	Fri Aug 13 08:58:07 2010 +0100
@@ -1355,7 +1355,8 @@
             // awesome.
             bool bHack =
                 getDisplay()->getWMAdaptor()->getWindowManagerName().EqualsAscii("Metacity") ||
-                getDisplay()->getWMAdaptor()->getWindowManagerName().EqualsAscii("compiz")
+                getDisplay()->getWMAdaptor()->getWindowManagerName().EqualsAscii("compiz") ||
+                getDisplay()->getWMAdaptor()->getWindowManagerName().EqualsAscii("Mutter")
                 ;
             if( nUserTime == 0 && bHack )
             {
